tools/integration_tests/util/operations/file_operations.go

```diff
@@ -642,3 +632,61 @@ func SizeOfFile(filepath string) (size int64, err error) {
 
 	return (*fstat).Size(), nil
 }
+
+// Creates a temporary file (name-collision-safe) in /tmp with given content.
+// If gzipCompress is true, output file is a gzip-compressed file.
+// Caller is responsible for deleting the created file when done using it.
+// Failure cases:
+// 1. os.CreateTemp() returned error or nil handle
+// 2. gzip.NewWriter() returned nil handle
+// 3. Failed to write the content to the created temp file
+func CreateLocalTempFile(content string, gzipCompress bool) (string, error) {
+	contentSize := len(content)
+
+	// create appropriate name template for temp file
+	filenameTemplate := "testfile-*.txt"
+	if gzipCompress {
+		filenameTemplate += ".gz"
+	}
+
+	// create a temp file
+	f, err := os.CreateTemp(TmpDirectory, filenameTemplate)
+	if err != nil {
+		return "", fmt.Errorf("failed to create tempfile for template %s: %w", filenameTemplate, err)
+	}
+	if f == nil {
+		return "", fmt.Errorf("nil file handle returned from os.CreateTemp")
+	}
+	defer CloseFile(f)
+	filepath := f.Name()
+
+	if gzipCompress {
+		w := gzip.NewWriter(f)
+		if w == nil {
+			return "", fmt.Errorf("failed to create gzip writer for file %s", filepath)
+		}
+		defer func() {
+			if err := w.Close(); err != nil {
+				fmt.Printf("failed to close gzip writer for file %s: %v", filepath, err)
```

Comment:
nit: ideally we should return this error and not just print it.

Reply:
We can't return anything in defer function and also returning a non nil error will result in test failure but not closing writer should not result in test failure.

Reply:
see my point at https://github.com/GoogleCloudPlatform/gcsfuse/pull/2442#discussion_r1760472483

Reply:
Resolving this as per discussion.